modules/widgets/internet-defense-league.php

```diff
@@ -0,0 +1,154 @@
+<?php /*
+
+**************************************************************************
+
+Plugin Name:  Internet Defense League
+Description:  Displays your support for the Internet Defense League.
+Author:       Automattic Inc.
+Author URI:   https://automattic.com/
+
+**************************************************************************/
+
+class Internet_Defense_League_Widget extends WP_Widget {
+
+	public $defaults = array();
+
+	public $variant;
+	public $variants = array();
+
+	public $campaign;
+	public $campaigns  = array();
+	public $no_current = true;
+
+	public $badge;
+	public $badges = array();
+
+	function __construct() {
+		parent::__construct(
+			'internet_defense_league_widget',
+			apply_filters( 'jetpack_widget_name_widget', esc_html__( 'Internet Defense League', 'jetpack' ) ),
+			array(
+				'description' => __( 'Show your support for the Internet Defense League.', 'internetdefenseleague' ),
+			)
+		);
+
+		// When enabling campaigns other than 'none' or empty, change $no_current to false above.
+		$this->campaigns = array(
+			''       => __( 'All current and future campaigns', 'internetdefenseleague' ),
+			// 'nsa' => __( 'NSA Protest on July 4th, 2013', 'internetdefenseleague' ),
+			'none'   => __( 'None, just display the badge please', 'internetdefenseleague' ),
+		);
+
+		$this->variants = array(
+			'banner' => __( 'Banner at the top of my site', 'internetdefenseleague' ),
+			'modal'  => __( 'Modal (Overlay Box)', 'internetdefenseleague' ),
+		);
+
+		$this->badges = array(
+			'shield_badge'   => __( 'Shield Badge', 'internetdefenseleague' ),
+			'super_badge'    => __( 'Super Badge', 'internetdefenseleague' ),
+			'side_bar_badge' => __( 'Red Cat Badge', 'internetdefenseleague' ),
+			'none'           => __( 'Don\'t display a badge (just the campaign)', 'internetdefenseleague' ),
+		);
+
+		$this->defaults = array(
+			'campaign' => key( $this->campaigns ),
+			'variant'  => key( $this->variants ),
+			'badge'    => key( $this->badges ),
+		);
+	}
+
+	public function widget( $args, $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults );
+
+		if ( 'none' != $instance['badge'] ) {
+			if ( ! isset( $this->badges[ $instance['badge'] ] ) ) {
+				$instance['badge'] = $this->defaults['badge'];
+			}
+			echo $args['before_widget'];
+			echo '<p><a href="https://internetdefenseleague.org/"><img src="' . esc_url( 'https://internetdefenseleague.org/images/badges/final/' . $instance['badge'] . '.png' ) . '" alt="Member of The Internet Defense League" style="max-width: 100%; height: auto;" /></a></p>';
+			echo $args['after_widget'];
+			do_action( 'jetpack_stats_extra', 'widget_view', 'internet_defense_league' );
+		}
+
+		if ( 'none' != $instance['campaign'] ) {
+			$this->campaign = $instance['campaign'];
+			$this->variant  = $instance['variant'];
+			add_action( 'wp_footer', array( $this, 'footer_script' ) );
+			do_action( 'jetpack_stats_extra', 'widget_view', 'internet_defense_league' );
```

Comment:
Instead of adding the same action twice, how about placing it at the end of the widget output, so it's always triggered when the widget is in use, and always triggered only once?